examples/more/external-project-add/cmake/Catch2_External.cmake

```diff
@@ -0,0 +1,49 @@
+include(ExternalProject)
+
+###############################################################################
+# Catch2
+#  - use commit fd9f5ac661f87335ecd70d39849c1d3a90f1c64d
+#    v2.13.1
+###############################################################################
+# The parent script needs to define the "GLOBAL_OUTPUT_PATH" variable,
+# which will be used as output directory for all *.lib, *.dll, *.a, *.so, *.pdb files.
+
+set(CATCH2_ROOT        ${GLOBAL_OUTPUT_PATH}/catch2)
+set(CATCH2_INCLUDE_DIR ${CATCH2_ROOT}/include)
+
+message(INFO "ExternalProject_Add() - catch2 - building ${CMAKE_BUILD_TYPE}; CMAKE_BINARY_DIR=${CMAKE_BINARY_DIR}")
+
+ExternalProject_Add(
+    CATCH2_external
```

Comment:
For me I'm not sure we should actually be encouraging the use of `ExternalProject_Add` for `Catch2` - I think it's actually far simpler to just use `FetchContent` instead (it's a newer feature in CMake and built on-top of `ExternalProject_Add`.

The Professional CMake book goes into more detail about the advantages of `ExternalProject_Add`, mainly that it can been builds isolated when needed. This can be useful in certain contexts but often isn't necessary. Having an example is still a great idea, I just think it could be paired back a bit to the essentials as in introduction.

Reply:
This is the "big" struggle for me right now - finding a way to have Catch2 working correctly.